tf2/include/tf2/buffer_core.h

```diff
@@ -109,7 +110,20 @@ class BufferCore
    * \param is_static Record this transform as a static transform.  It will be good across all time.  (This cannot be changed after the first call.)
    * \return True unless an error occured
    */
-  bool setTransform(const geometry_msgs::TransformStamped& transform, const std::string & authority, bool is_static = false);
+  bool setTransform(const geometry_msgs::msg::TransformStamped& transform, const std::string & authority, bool is_static = false)
+  {
+    tf2::Transform tf2_transform(tf2::Quaternion(transform.transform.rotation.w,
+                                                 transform.transform.rotation.x,
+                                                 transform.transform.rotation.y,
+                                                 transform.transform.rotation.z),
+                                 tf2::Vector3(transform.transform.translation.x,
+                                              transform.transform.translation.y,
+                                              transform.transform.translation.z));
+    TimePoint time_point(std::chrono::nanoseconds(transform.header.stamp.nanosec) +
+        std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::seconds(transform.header.stamp.sec)));
+    return setTransformImpl(tf2_transform, transform.header.frame_id, transform.child_frame_id,
+                            time_point, authority, is_static);
+  }
```

Comment:
Why was the implementation moved here? Does it need to be `inline` since it's not a template?